### PR TITLE
[Brownfield] Structure changes

### PR DESCRIPTION
**Pull request by annakocheshkova**
Made several important code structure changes. Among them are the following:

* Got rid of `JsonObject` usage in `CodePushUpdateManger`, which led to the subset of changes:
   * Introduced a new data contract representing package info (`CodePushPackageInfo`);
   * Got rid of unnecessary `CodePushUpdateManagerDeserializer`;
   * Introduced two new utils methods allowing to simplify json conversion.
* Added an interface for platform-specific utils with one example of its usage;
* Added a `tearDown` method for file tests to clear used folders.

... and updated tests accordingly.

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -44,6 +44,12 @@ android {
             returnDefaultValues = true
         }
     }
+    packagingOptions {
+        exclude 'META-INF/LICENSE.txt'
```
Comment (sergey-akhalkov) [inline android/app/build.gradle]: Do we really need this fakes (please let me know if they are not fakes) to be merged?

File: android/app/src/androidTest/java/com/microsoft/codepush/common/UpdateManagerTest.java [hunk 2]
```diff
@@ -102,11 +95,19 @@ public void setUp() throws Exception {
     @Test
     public void fullWorkflowTest() throws Exception {
         codePushUpdateManager.clearUpdates();
-        executeFullWorkflow(packageObject, codePushUpdateManager, FULL_PACKAGE_HASH, FULL_PACKAGE_URL);
-        executeFullWorkflow(packageObject, codePushUpdateManager, DIFF_PACKAGE_HASH, DIFF_PACKAGE_URL);
-        CodePushLocalPackage codePushPreviousPackage = codePushUpdateManagerDeserializer.getPreviousPackage();
-        CodePushLocalPackage codePushCurrentPackage = codePushUpdateManagerDeserializer.getCurrentPackage();
-        CodePushLocalPackage codePushPackage = codePushUpdateManagerDeserializer.getPackage(DIFF_PACKAGE_HASH);
+        executeFullWorkflow(codePushUpdateManager, FULL_PACKAGE_HASH, FULL_PACKAGE_URL);
```
Comment (sergey-akhalkov) [inline android/app/src/androidTest/java/com/microsoft/codepush/common/UpdateManagerTest.java]: What is actually `FULL_` prefix means?

File: android/app/src/main/java/com/microsoft/codepush/common/datacontracts/CodePushLocalPackage.java [hunk 3]
```diff
@@ -16,6 +16,12 @@
     @SerializedName("isPending")
     private boolean isPending;
 
+    /**
+     * Indicates the path to the application entry point.
+     */
+    @SerializedName("appEntryPoint")
+    private String appEntryPoint;
```
Comment (sergey-akhalkov) [inline android/app/src/main/java/com/microsoft/codepush/common/datacontracts/CodePushLocalPackage.java]: Please be careful with properties names changes of local manifests and etc. I mean in terms of backward compatibility.

File: android/app/src/test/java/com/microsoft/codepush/common/UpdateManagerCommonTest.java [hunk 4]
```diff
@@ -63,7 +62,7 @@ public void downloadFailsIfDeleteNewUpdateFolderPathFails() throws Exception {
         mockStatic(FileUtils.class);
         PowerMockito.doThrow(new IOException()).when(FileUtils.class, "deleteDirectoryAtPath", anyString());
         PowerMockito.doReturn(true).when(FileUtils.class, "fileAtPathExists", anyString());
-        codePushUpdateManager.downloadPackage(mock(JSONObject.class), mock(DownloadProgressCallback.class), mock(PackageDownloader.class));
+        codePushUpdateManager.downloadPackage("", mock(PackageDownloader.class));
```
Comment (sergey-akhalkov) [inline android/app/src/test/java/com/microsoft/codepush/common/UpdateManagerCommonTest.java]: Could you please clarify why we have changed `mock(JSONObject.class), mock(DownloadProgressCallback.class)` to an empty string?

File: android/app/build.gradle (hunk 1 above)
Comment (annakocheshkova) [reply android/app/build.gradle]: Yes, this is required for gradle 2.2.3 not to throw a merge error.

File: android/app/src/test/java/com/microsoft/codepush/common/UpdateManagerCommonTest.java (hunk 4 above)
Comment (annakocheshkova) [reply android/app/src/test/java/com/microsoft/codepush/common/UpdateManagerCommonTest.java]: @sergey-akhalkov, signature of the method has changed to take a String with package hash.

File: android/app/src/androidTest/java/com/microsoft/codepush/common/UpdateManagerTest.java (hunk 2 above)
Comment (annakocheshkova) [reply android/app/src/androidTest/java/com/microsoft/codepush/common/UpdateManagerTest.java]: @sergey-akhalkov, have added the explanation.

**Pull request merged**